src/views/Nft/market/hooks/useNftOwner.tsx

```diff
@@ -1,13 +1,15 @@
 import { useEffect, useState } from 'react'
-import { useErc721CollectionContract } from 'hooks/useContract'
 import { NftToken } from 'state/nftMarket/types'
+import { getErc721CollectionContract } from 'utils/contractHelpers'
+import useActiveWeb3React from 'hooks/useActiveWeb3React'
 
 const NOT_ON_SALE_SELLER = '0x0000000000000000000000000000000000000000'
 
 const useNftOwner = (nft: NftToken) => {
+  const { library } = useActiveWeb3React()
   const [owner, setOwner] = useState(null)
   const [isLoadingOwner, setIsLoadingOwner] = useState(true)
-  const collectionContract = useErc721CollectionContract(nft.collectionAddress)
+  const collectionContract = getErc721CollectionContract(library, nft.collectionAddress)
```

Comment:
I think we need to memo it.
maybe we can change the `useErc721CollectionContract` and use `utils/getProviderOrSigner`

Reply:
Done